zerver/lib/retention.py

```diff
@@ -42,41 +43,44 @@
 ]
 
 @transaction.atomic(savepoint=False)
-def move_rows(base_model: Model, raw_query: str, src_db_table: str='', returning_id: bool=False,
-              **kwargs: Any) -> List[int]:
-    if not src_db_table:
+def move_rows(
+    base_model: Model,
+    raw_query: Composable,
+    *,
+    src_db_table: Optional[str]=None,
+    returning_id: bool=False,
+    **kwargs: Composable,
+) -> List[int]:
+    if src_db_table is None:
         # Use base_model's db_table unless otherwise specified.
         src_db_table = base_model._meta.db_table
 
-    src_fields = ["{}.{}".format(src_db_table, field.column) for field in base_model._meta.fields]
-    dst_fields = [field.column for field in base_model._meta.fields]
+    src_fields = [
+        SQL("{}.{}").format(Identifier(src_db_table), Identifier(field.column))
```

Comment:
Wouldn't `Identifier(src_db_table, field.column)`  be better here? see second example in https://www.psycopg.org/docs/sql.html#psycopg2.sql.Identifier

Reply:
Yeah, that does seem better.